### ISSUE TITLE
Crashes when deleting sections

### ISSUE THREAD

**Issue by ankushdhawan** [fuzz-productions/FZAccordionTableView#4]
Hello Sir,

When i delete row from table  and set multiple section open properties off .When i select any header app becomes crash.

Thanks and Regards
Ankush

**Comment by kgaidis** [comment]
Apologize for late response. 

I could not reproduce the issue. Under Example folder, I added "Testing_Example" project in which you can use the swipe-to-delete feature to delete rows. If you could describe your problem in a step-by-step fashion through that Example, that would be very helpful.

**Comment by mrishi** [comment]
Hi,

I too experienced the issue. Your 'Testing Example' does not cover it. 

To reproduce the issue :
1. Prepare multiple sections with different number of rows in each.
2. Add a button on the sections to allow for deleting the section.
3. If  a section is deleted :
- The next section doesn't get selected and open automatically ( even if the flag for "Keep one section Open" is selected.
- Clicking on the next/previous collapsed section results in a crash.

**Comment by ankushdhawan** [comment]
i resolve this crash by doing some changes below function in FZAccordionTableView
- (void)autoCollapseAllSectionsExceptSection:(NSInteger)section {
  // Get all of the sections that we need to close
  int sectionCount = [self numberOfSections];
  NSLog(@"%d",sectionCount);
  NSMutableSet *sectionsToClose = [[NSMutableSet alloc] init];
  for (NSNumber *openedSection in self.openedSections) {
      if (openedSection.integerValue <  self.numberOfSections) {
          if (openedSection.integerValue != section && ![self isAlwaysOpenedSection:openedSection.integerValue]) {
              [sectionsToClose addObject:openedSection];
          }
      }
  }
  
  // Close the found sections
  for (NSNumber *sectionToClose in sectionsToClose) {
      if (sectionToClose.integerValue <  self.numberOfSections) {
          if ([self.subclassDelegate respondsToSelector:@selector(tableView:willCloseSection:withHeader:)]) {
              [self.subclassDelegate tableView:self willCloseSection:sectionToClose.integerValue withHeader:[self headerViewForSection:sectionToClose.integerValue]];
          }
      }
      // Change animations based off which sections are closed
      UITableViewRowAnimation closeAnimation = UITableViewRowAnimationTop;
      if (section < sectionToClose.integerValue) {
          closeAnimation = UITableViewRowAnimationBottom;
      }
      if (self.enableAnimationFix) {
          if ((sectionToClose.integerValue == self.numOfRowsForSection.count - 1 ||
               sectionToClose.integerValue == self.numOfRowsForSection.count - 2) &&
              !self.allowsMultipleSelection) {
              closeAnimation = UITableViewRowAnimationFade;
          }
      }
  
  ```
  // Delete the cells for section that is closing
  NSArray *indexPathsToDelete = [self getIndexPathsForSection:sectionToClose.integerValue];
  [self removeOpenedSection:sectionToClose.integerValue];
  
  [self beginUpdates];
  [CATransaction setCompletionBlock:^{
      if ([self.subclassDelegate respondsToSelector:@selector(tableView:didCloseSection:withHeader:)]) {
          [self.subclassDelegate tableView:self didCloseSection:sectionToClose.integerValue withHeader:[self headerViewForSection:sectionToClose.integerValue]];
      }
  }];
  [self deleteRowsAtIndexPaths:indexPathsToDelete withRowAnimation:closeAnimation];
  [self endUpdates];
  ```
  
  }
  }

**Comment by kgaidis** [comment]
OK thank you, I'll look into that. I only tested the deletion of rows, I did not test the deletion of sections and I could see how that could cause problems. Hope to have this fixed by the end of the week for sure.

**Comment by kgaidis** [comment]
Just to give an update, it looks like this might require internal data structure changes, which means quite a bit of refactoring. In some sense, this wouldn't be _too_ difficult to do inefficiently, but I want to try to find more efficient solutions. 

The problem is exacerbated by the fact that UITableView has great support for **rows**, but doesn't have great support for **headers**. For example, a dynamic function like sectionForHeaderView doesn't exist. 

I am working on fixing this though!

**Comment by mrishi** [comment]
Thanks Chris,

I moved to TLIndexPathTools, which provides section-collapse feature along
with many other features. This required big changes at the model level but
allows for more consistency and tighter-integration with lesser code.

You may want to check if FZAccordionTableview can share some code.

For the dynamic function on header, will viewForHeaderInSection and
willDisplayHeaderForSection not help?

Thanks,
Rishi

**Comment by kgaidis** [comment]
Fixed in 0.2.0.
